src/main/java/taskmanager/ConditionHandlers.java

```diff
@@ -0,0 +1,175 @@
+                splittedMessage[i] = splittedMessage[i].trim();
+            }
+            // user did not key in a task for the event
+            if (splittedMessage[1].startsWith("/by")) {
+                throw new InvalidInputException();
+            }
+            String[] doubleSplittedMessage = splittedMessage[1].split("/by");
+            for (int j = 0; j < doubleSplittedMessage.length; j += 1) {
+                doubleSplittedMessage[j] = doubleSplittedMessage[j].trim();
+            }
+            // user did not key in a start date for the event or did not key in the task or both
+            if (doubleSplittedMessage.length == 0 ||  doubleSplittedMessage.length == 1){
+                throw new InvalidInputException();
+            }
+            taskList[taskCounter] = new Task(doubleSplittedMessage[0]);
+            taskList[taskCounter].setTaskType("deadline");
+            taskList[taskCounter].setEndDate(doubleSplittedMessage[1]);
+            taskCounter += 1;
+            Messages.addDeadlineMessage(taskList[taskCounter - 1].getTaskType(),
+                    taskList[taskCounter - 1].getStatusIcon(), taskList[taskCounter - 1].getDescription(),
+                    taskList[taskCounter - 1].getEndDate(), taskCounter);
+            return taskCounter;
+        } catch (InvalidInputException e) {
+            Messages.typoErrorMessage();
+            return taskCounter;
+        }
+    }
+    public static int addEventTaskToList(String receivedMessage, Task[] taskList, int taskCounter){
+        try {
+            receivedMessage = receivedMessage.trim();
+            if (!receivedMessage.startsWith("event") || !receivedMessage.contains("/from")
+                    || !receivedMessage.contains("/to")) { // input does not start with event
+                throw new InvalidInputException();
+            }
+            String[] splittedMessage = receivedMessage.split("event ");
+            for (int i = 0; i < splittedMessage.length; i += 1) {
+                splittedMessage[i] = splittedMessage[i].trim();
+            }
+            // user did not key in a task for the event or did not key in a start date
+            if (splittedMessage[1].startsWith("/from") || splittedMessage[1].startsWith("/to")) {
+                throw new InvalidInputException();
+            }
+            String[] doubleSplittedMessage = splittedMessage[1].split("/from");
+            for (int j = 0; j < doubleSplittedMessage.length; j += 1) {
+                doubleSplittedMessage[j] = doubleSplittedMessage[j].trim();
+            }
+            if (doubleSplittedMessage[1].startsWith("/to")) { // user did not key in a start date for the event
+                throw new InvalidInputException();
+            }
+            String[] tripleSplittedMessage = doubleSplittedMessage[1].split("/to");
+            if (tripleSplittedMessage.length == 0 || tripleSplittedMessage.length == 1) {
```

Comment:
This is a complex statement. Try to break it down into sub-expressions and store the results in a boolean variable.